File: allure-junit4/src/test/java/io/qameta/allure/examples/junit4/AllureFixtureTest.java
package io.qameta.allure.examples.junit4;

import org.junit.*;

import static io.qameta.allure.Allure.step;

public class AllureFixtureTest {

    @BeforeClass
    public static void beforeClass() {
        step("Step inside beforeClass");
    }

    @Before
    public void before() {
        step("Step inside before");
    }

    @Test
    public void allureFixtureTest() {
        step("Step inside allureFixtureTest");
    }

    @Test
    public void allureFixtureTest2() {
        step("Step inside allureFixtureTest2");
    }

    @After
    public void after() {
        step("Step inside after");
    }

    @AfterClass
    public static void afterClass() {
        step("Step inside afterClass");
    }

}

File: allure-javascript/allure-javascript-jest/__tests__/allure-fixture.test.js
// beforeAll(() => {
//   allure.step("Step inside before all", () => {});
// });
//
// afterAll(() => {
//   allure.step("Step inside after all", () => {});
// });

beforeEach(() => {
  allure.step("Step inside before each", () => {});
});

afterEach(() => {
  allure.step("Step inside after each", () => {});
});

describe("test", () => {
  beforeAll(() => {
    allure.step("Step inside describe before all", () => {});
  });
  afterAll(() => {
    allure.step("Step inside describe after all", () => {});
  });
  beforeEach(() => {
    allure.step("Step inside describe before each", () => {});
  });
  afterEach(() => {
    allure.step("Step inside describe after each", () => {});
  });
  test("Allure fixture test 1", () => {
    allure.step("Simple step", () => {});
  });
  test("Allure fixture test 2", () => {
    allure.step("Simple step", () => {});
  });
});
